src/View/CastingService.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace SilverStripe\View;
+
+use LogicException;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Model\ArrayData;
+use SilverStripe\Model\List\ArrayList;
+use SilverStripe\Model\ModelData;
+use SilverStripe\ORM\FieldType\DBBoolean;
+use SilverStripe\ORM\FieldType\DBFloat;
+use SilverStripe\ORM\FieldType\DBInt;
+use SilverStripe\ORM\FieldType\DBText;
+
+class CastingService
+{
+    use Injectable;
+
+    /**
+     * Cast a value to the relevant object (usually a DBField instance) for use in the view layer.
+     *
+     * @param null|array|ModelData $source Where the data originates from. This is used both to check for casting helpers
+     * and to help set the value in cast DBField instances.
+     * @param bool $strict If true, an object will be returned even if $data is null.
+     */
+    public function cast(mixed $data, null|array|ModelData $source = null, string $fieldName = '', bool $strict = false): ?object
+    {
+        // null is null - we shouldn't cast it to an object, because that makes it harder
+        // for downstream checks to know there's "no value".
+        if (!$strict && $data === null) {
+            return null;
+        }
+
+        // Assume anything that's an object is intentionally using whatever class it's using
```

Comment:
Move this above the `if (!$strict && $data === null) {` check

Reply:
But it's specifically intended to show why we're returning the value directly if it's an object. I'm not sure why you want it moved to the bit that talks about null?

Reply:
Reads a little better to me if the is_object() check is before the is_null() check, though it really doesn't matter so just leave it

Reply:
Oh you meant move the whole check! I can do that, that's fine. It looked like you meant to move just the comment lol

Reply:
Moved the object check to be before the null check.